src/node/txdownloadman_impl.h

```diff
@@ -128,11 +128,28 @@ class TxDownloadManagerImpl {
 
     TxDownloadManagerImpl(const TxDownloadOptions& options) : m_opts{options} {}
 
+    struct PeerInfo {
+        /** Information relevant to scheduling tx requests. */
+        const TxDownloadConnectionInfo m_connection_info;
+
+        PeerInfo(const TxDownloadConnectionInfo& info) : m_connection_info{info} {}
+    };
+
+    /** Information for all of the peers we may download transactions from. This is not necessarily
+     * all peers we are connected to (no block-relay-only and temporary connections). */
+    std::map<NodeId, PeerInfo> m_peer_info;
+
+    /** Number of wtxid relay peers we have. */
+    uint32_t m_num_wtxid_peers{0};
```

Comment:
6fbb2463840f47f9f0632431e98a68f6747805ea

This variable is kinda synchronous with `m_peer_info`.
Would be unfortunate to decouple it accidentally in future code. Perhaps comment on it?

Reply:
Added a comment that this specifically refers to peers in the map.

I also added a doxygen comment to the `TxDownloadManager` definition saying that this class isn't thread-safe and needs to be externally synced.